ludwig/schema/hyperopt/scheduler.py

```diff
@@ -0,0 +1,509 @@
+# To prevent direct dependency on ray import, the following static key stores are duplicated:
+
+# from ray.tune.schedulers import SCHEDULER_IMPORT
+# https://github.com/ray-project/ray/blob/137a1b12c3b31a3622fa5f721a05a64e9b559b05/python/ray/tune/schedulers/__init__.py#L28
+
+# from ray.tune.result import DEFAULT_RESULT_KEYS
+# Taken from https://github.com/ray-project/ray/blob/137a1b12c3b31a3622fa5f721a05a64e9b559b05/python/ray/tune/result.py
+TRAINING_ITERATION = "training_iteration"
+TIME_TOTAL_S = "time_total_s"
+TIMESTEPS_TOTAL = "timesteps_total"
+MEAN_ACCURACY = "mean_accuracy"
+MEAN_LOSS = "mean_loss"
+DEFAULT_RESULT_KEYS = (TRAINING_ITERATION, TIME_TOTAL_S, TIMESTEPS_TOTAL, MEAN_ACCURACY, MEAN_LOSS)
+
+# from ray.tune.result import DEFAULT_METRIC
+RAY_TUNE_DESULT_DEFAULT_METRIC = "_metric"
+# ----------------------------------------------------------------------------------------------------------------------
+
+scheduler_config_registry = Registry()
+
+
+def register_scheduler_config(name: str):
+    def wrap(scheduler_config: BaseSchedulerConfig):
+        scheduler_config_registry[name] = scheduler_config
+        return scheduler_config
+
+    return wrap
+
+
+@dataclass
+class BaseSchedulerConfig(schema_utils.BaseMarshmallowConfig, ABC):
+    """Base class for schedulers.
+
+    Not meant to be used directly.
+    """
+
+    type: str
+    """Name corresponding to a scheduler in `ludwig.schema.hyperopt.scheduler.scheduler_registry`.
+       Technically mutable, but attempting to load a derived scheduler with `type` set to a mismatched value will
+       result in a `ValidationError`."""
+
+
+# Field aliases to cut down on code reuse:
+def metric_alias(default=None):
+    return schema_utils.StringOptions(
+        options=list(DEFAULT_RESULT_KEYS) + [RAY_TUNE_DESULT_DEFAULT_METRIC],
+        default=default,
+        description=(
+            "The training result objective value attribute. Stopping procedures will use this attribute. If None but a "
+            "mode was passed, the ray.tune.result.DEFAULT_METRIC will be used per default."
+        ),
+    )
+
+
+def time_attr_alias(default=TRAINING_ITERATION):
+    return schema_utils.StringOptions(
+        options=list(DEFAULT_RESULT_KEYS),
+        default=default,
+        allow_none=False,
+        description=(
+            "A training result attr to use for comparing time. Note that you can pass in something non-temporal such as"
+            " training_iteration as a measure of progress, the only requirement is that the attribute should increase "
+            "monotonically."
+        ),
+    )
+
+
+def max_t_alias(default=100):
+    return schema_utils.PositiveInteger(
+        default=default,
+        description=(
+            "max time units per trial. Trials will be stopped after max_t time units (determined by time_attr) have "
+            "passed."
+        ),
+    )
+
+
+@dataclass
+class CommonSchedulerOptions:
```

Comment:
@arnavgarg1 This is a class to hold some common scheduler options, re: your comment (https://github.com/ludwig-ai/ludwig/pull/2346#discussion_r1028045715). I also defined a couple of aliases to reduce verbosity.

Reply:
Awesome, this is neat